src/imgui/ImGuiMachine.cc

```diff
@@ -271,14 +259,124 @@ void ImGuiMachine::showMenu(MSXMotherBoard* motherBoard)
+							if (ImGui::Selectable(info.displayName.c_str())) {
+								reactor.getDefaultMachineSetting().setValue(TclObject(info.configName));
+								// clear the default setup setting to avoid loading that at startup
+								reactor.getDefaultSetupSetting().setValue(TclObject(""));
+							}
+							if (info.configName == currentDefault && ImGui::IsWindowAppearing()) ImGui::SetScrollHereY();
+							if (ImGui::IsItemHovered(ImGuiHoveredFlags_DelayNormal | ImGuiHoveredFlags_NoSharedDelay | ImGuiHoveredFlags_Stationary)) {
+								im::ItemTooltip([&]{
+									printConfigInfo(info);
+								});
+							}
+						});
+					}
+				});
+			});
+			ImGui::RadioButton("Load a setup", &startType, StartType::SETUP);
+			HelpMarker("Select this option to load the specified setup when openMSX starts up. "
+				"Note that in case the setup cannot be loaded, the machine shown at the other "
+				"option will be loaded after all.");
+			im::DisabledIndent(startType != StartType::SETUP, [&] {
+				if (!setupSettingsOpen) {
+					setups = Reactor::getSetups();
+				}
+				auto currentDefault = startType == StartType::SETUP ? reactor.getDefaultSetupSetting().getString() : previousDefaultSetup;
+				auto showSetup = [&](const std::string& setup) {
+					if (ImGui::Selectable(setup.c_str())) {
+						reactor.getDefaultSetupSetting().setValue(TclObject(setup));
+					}
+					if (setup == currentDefault && ImGui::IsWindowAppearing()) ImGui::SetScrollHereY();
+					if (ImGui::IsItemHovered(ImGuiHoveredFlags_DelayNormal | ImGuiHoveredFlags_NoSharedDelay | ImGuiHoveredFlags_Stationary)) {
+						if (previewSetup.name != setup) {
+							try {
+								previewSetup.fullName = userDataFileContext(Reactor::SETUP_DIR).resolve(tmpStrCat(setup, Reactor::SETUP_EXTENSION));
+								loadPreviewSetup();
+							} catch (FileException&) {
+								manager.executeDelayed([this] {
+									previewSetup.motherBoard.reset();
+								});
+							}
+						}
+						im::ItemTooltip([&]{
+							if (previewSetup.motherBoard) {
+								showSetupOverview(*previewSetup.motherBoard, ViewMode::NO_CONTROLS);
+							} else {
+								showNonExistingPreview();
+							}
+						});
+					}
+				};
+
+				im::Combo("##defaultSetup", currentDefault.c_str(), [&]{
+					for (auto& setup: setups) {
+						showSetup(setup);
+					}
+					auto currentSaveAtExitName = manager.getReactor().getSaveSetupAtExitNameSetting().getString();
+					if (!contains(setups, currentSaveAtExitName)) {
+						showSetup(std::string(currentSaveAtExitName));
+					}
+				});
+			});
+			auto& defaultSetupSetting = reactor.getDefaultSetupSetting();
+			if (oldStartType != startType) {
+				if (startType == StartType::MACHINE) {
+					previousDefaultSetup = defaultSetupSetting.getValue().getString();
+					defaultSetupSetting.setValue(TclObject(""));
+				} else {
+					auto currentDefaultSetup = defaultSetupSetting.getValue().getString();
+					if (currentDefaultSetup.empty()) {
```

Comment:
Why this check? We know we're switching from machine->setup. So the "default_setup" setting *must* be empty, no?

Reply:
I guess I was overly cautious here. Thanks, will change.